modules/login/php/endpoints/authentication.class.inc

```diff
@@ -0,0 +1,453 @@
+            $response['status'] = 'success';
+        } else if ($values['command'] === 'request') {
+            // request - account request by user.
+            $response           = $this->requestAccount($values);
+            $response['status'] = 'success';
+        } else if ($values['command'] === 'login') {
+            // login - user attempting to login.
+            $response = $this->login($values);
+        } else if ($values['command'] === 'expired') {
+            // login - expired password update
+            $response = $this->expiredPassword($values);
+        }
+
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Initialize setup.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function initializeSetup(array $values) : array
+    {
+        // Login Setup.
+        $loginData = array();
+        $login     = $_SESSION['State']->getProperty('login');
+
+        // Previous login failed.
+        if (!empty($login->_lastError)) {
+            $loginData['error'] = $login->_lastError;
+        }
+        $config      =& \NDB_Config::singleton();
+        $study_links = $config->getExternalLinks('Studylinks');
+        foreach ($study_links as $label => $url) {
+            $windowName           = md5($url);
+            $loginData['links'][] = array(
+                'url'        => $url,
+                'label'      => $label,
+                'windowName' => $windowName,
+            );
+        }
+        $loginData['description'] = html_entity_decode(
+            $config->getSetting('StudyDescription')
+        );
+        $loginData['title']       = $config->getSetting('title');
+        $loginData['logo']        = $config->getSetting('studylogo');
+        $values['login']          = $loginData;
+
+        // Request Account Setup.
+        $requestAccountData = array();
+        $reCAPTCHAPrivate   = $config->getSetting('reCAPTCHAPrivate');
+        $reCAPTCHAPublic    = $config->getSetting('reCAPTCHAPublic');
+        // reCAPTCHA if both private and public keys are set
+        if ($reCAPTCHAPrivate && $reCAPTCHAPublic) {
+            $requestAccountData['captcha'] = $reCAPTCHAPublic; // captcha key.
+        }
+        $requestAccountData['site'] = \Utility::getSiteList();
+        $values['requestAccount']   = $requestAccountData;
+
+        return $values;
+    }
+
+    /**
+     * Reset password.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function resetPassword(array $values) : array
+    {
+        // Always claim success to avoid providing user info to attackers.
+        $values['success'] = 'Password reset. '
+            . 'You should receive an email within a few minutes.';
+
+        // Create the user object.
+        $username = $values['username'];
+        $user     =& \User::singleton($username);
```

Comment:
Factory

Reply:
Hi @johnsaigle can you make a user with factory from a username?